src/canvas-extractor/canvas_extractor/__main__.py

```diff
@@ -52,7 +48,29 @@
 start_date: str = ""
 end_date: str = ""
 
+results_store: Dict[str, Tuple[list, DataFrame]] = {}
```

Comment:
The type of results_store needs to be loosened because on line 151, udm_assignments_df is a Dict[str, DataFrame], not a DataFrame:
`results_store: Dict[str, Tuple[list, Union[DataFrame, Dict[str, DataFrame]]]] = {}`

Reply:
BTW, looking forward to Python 3.10 release, which includes TypeScript-like Union operator:
https://www.python.org/dev/peps/pep-0604/
Turns that into a more reasonable:
`results_store: Dict[str, Tuple[list, DataFrame | Dict[str, DataFrame]]] = {}`

Reply:
✔️